worker/bootstrap/manifold.go

```diff
@@ -75,19 +112,40 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 			config.StateName,
 			config.ObjectStoreName,
 			config.BootstrapGateName,
+			config.ServiceFactoryName,
 		},
 		Start: func(context dependency.Context) (worker.Worker, error) {
 			if err := config.Validate(); err != nil {
 				return nil, errors.Trace(err)
 			}
 
+			var bootstrapUnlocker gate.Unlocker
+			if err := context.Get(config.BootstrapGateName, &bootstrapUnlocker); err != nil {
+				return nil, errors.Trace(err)
+			}
+
 			var a agent.Agent
 			if err := context.Get(config.AgentName, &a); err != nil {
 				return nil, err
 			}
 
-			var bootstrapUnlocker gate.Unlocker
-			if err := context.Get(config.BootstrapGateName, &bootstrapUnlocker); err != nil {
+			// If the controller application exists, then we don't need to
```

Comment:
See here, `RequiresBootstrap` no longer checks if the app exists, so this becomes misleading.